### PR TITLE
MONIT-30269: changes to ingested "error_name" instead of "exception" for summary

### PR DESCRIPTION
**Pull request by panjwanipr-vmware**
- changes to ingested "error_name" instead of "exception" for summary
- User can provide custom exception and log level tags names for alert summary page.

### CODE REVIEW

File: proxy/src/main/java/com/wavefront/agent/ProxyConfig.java [hunk 1]
```diff
@@ -1309,6 +1309,22 @@ public class ProxyConfig extends Configuration {
               + "`service`. Default: none")
   String customServiceTags = "";
 
+  @Parameter(
+      names = {"--customExceptionTags"},
+      description =
+          "Comma separated list of log tag "
+              + "keys that should be treated as the exception in Wavefront in the absence of a "
+              + "tag named `exception`. Default: none")
```
Comment (amitw-vmware) [inline proxy/src/main/java/com/wavefront/agent/ProxyConfig.java:1317]: defaults should be exception and error_name.

File: proxy/src/main/java/com/wavefront/agent/ProxyConfig.java [hunk 2]
```diff
@@ -1309,6 +1309,22 @@ public class ProxyConfig extends Configuration {
...
+  String customExceptionTags = "";
+
+  @Parameter(
+      names = {"--customLevelTags"},
+      description =
+          "Comma separated list of log tag "
+              + "keys that should be treated as the log level in Wavefront in the absence of a "
+              + "tag named `level`. Default: none")
```
Comment (amitw-vmware) [inline proxy/src/main/java/com/wavefront/agent/ProxyConfig.java:1325]: default should be level and log_level

Comment (panjwanipr-vmware) [reply proxy/src/main/java/com/wavefront/agent/ProxyConfig.java]: Done Changes.

File: proxy/src/main/java/com/wavefront/agent/ProxyConfig.java [hunk 3]
```diff
@@ -1309,6 +1309,22 @@ public class ProxyConfig extends Configuration {
...
+  String customExceptionTags = "";
+
+  @Parameter(
+      names = {"--customLevelTags"},
+      description =
+          "Comma separated list of log tag "
+              + "keys that should be treated as the log level in Wavefront in the absence of a "
+              + "tag named `level`. Default: level, log_level")
+  String customLevelTags = "";
+
```
Comment (xuranchen) [inline proxy/src/main/java/com/wavefront/agent/ProxyConfig.java:1327]: Theres a section later in this file around line 2200 that adds logic for passing in a config file. Can you add these 2 variables there as well.

Comment (panjwanipr-vmware) [reply proxy/src/main/java/com/wavefront/agent/ProxyConfig.java:1327]: Do i need to add logic for level in **verifyAndInit()** method ?

Comment (panjwanipr-vmware) [reply proxy/src/main/java/com/wavefront/agent/ProxyConfig.java:1327]: Thanks @xuranchen , That was missed while manually merging code to fork. I have updated changes.

**Comment by panjwanipr-vmware** [comment]
@xuranchen Please review changes.

File: proxy/src/main/java/com/wavefront/agent/ProxyConfig.java (hunk 3 above)
Comment (xuranchen) [reply proxy/src/main/java/com/wavefront/agent/ProxyConfig.java:1327]: I'm talking about this function
https://github.com/wavefrontHQ/wavefront-proxy/blob/a23233f527a5bfde9072edc8906186ba30cce786/proxy/src/main/java/com/wavefront/agent/ProxyConfig.java#L2269

Can you add the 2 variables here.

**Comment by joannak-vmware** [comment]
LGTM, will approve PR after QA tested.

**Comment by joannak-vmware** [comment]
Jenkins build job failing due to ProxyConfig.java format errors. Please run `mvn clean install -DskipTests` locally to automatically reformat code

**Review by xuranchen — APPROVED** [review]
LGTM

**Review by gangadharaswamy — APPROVED** [review]

**Review by joannak-vmware — APPROVED** [review]
LGTM

**Review by xuranchen — APPROVED** [review]
LGTM

**Review by sbhakta-vmware — APPROVED** [review]
lgtm

**Pull request merged**